lang/java/avro/src/main/java/org/apache/avro/file/DeflateCodec.java

```diff
@@ -81,16 +95,15 @@ public ByteBuffer decompress(ByteBuffer data) throws IOException {
     ByteArrayOutputStream baos = getOutputBuffer(data.remaining());
     InflaterOutputStream ios = new InflaterOutputStream(baos, getInflater());
     writeAndClose(data, ios);
-    ByteBuffer result = ByteBuffer.wrap(baos.toByteArray());
+    ByteBuffer result = ByteBuffer.wrap(((Buffer) baos).getData(), 0, baos.size());
     return result;
   }
 
   private void writeAndClose(ByteBuffer data, OutputStream to) throws IOException {
-    byte[] input = data.array();
-    int offset = data.arrayOffset() + data.position();
-    int length = data.remaining();
     try {
-      to.write(input, offset, length);
+      WritableByteChannel channel = Channels.newChannel(to);
+
+      channel.write(data);
```

Comment:
@nandorKollar would it be possible to not include this change to address this issue?